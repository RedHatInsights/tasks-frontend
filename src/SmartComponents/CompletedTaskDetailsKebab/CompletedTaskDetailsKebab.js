import React, { useState } from 'react';
import propTypes from 'prop-types';
import {
  Dropdown,
  DropdownItem,
  DropdownList,
  MenuToggle,
} from '@patternfly/react-core';
import { EllipsisVIcon } from '@patternfly/react-icons';

const CompletedTaskDetailsKebabItems = ({ status, setModalOpened }) => {
  //let type = status === 'Running' ? 'cancel' : 'delete';
  let type = 'delete';

  return (
    <DropdownItem
      aria-label={`${type} task menu item`}
      data-testid={`${type}-task-kebab-button`}
      key={`${type}-task`}
      component="button"
      data-ouia-component-id={`${type}-task-dropdown-item`}
      onClick={() => setModalOpened(true)}
      isDisabled={status !== 'Completed'}
    >
      {type[0].toUpperCase() + type.slice(1)}
    </DropdownItem>
  );
};

CompletedTaskDetailsKebabItems.propTypes = {
  setModalOpened: propTypes.func,
  status: propTypes.string,
};

const CompletedTaskDetailsKebab = ({ status, setModalOpened }) => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <Dropdown
      onSelect={() => setIsOpen(false)}
      toggle={(toggleRef) => (
        <MenuToggle
          ref={toggleRef}
          id="executed-task-kebab"
          aria-label="Task details menu toggle"
          variant="plain"
          onClick={() => setIsOpen(!isOpen)}
          isExpanded={isOpen}
        >
          <EllipsisVIcon />
        </MenuToggle>
      )}
      isOpen={isOpen}
      shouldFocusToggleOnSelect
      popperProps={{ position: 'right' }}
    >
      <DropdownList>
        <CompletedTaskDetailsKebabItems
          status={status}
          setModalOpened={setModalOpened}
        />
      </DropdownList>
    </Dropdown>
  );
};

CompletedTaskDetailsKebab.propTypes = {
  setModalOpened: propTypes.func,
  status: propTypes.string,
};

export default CompletedTaskDetailsKebab;
