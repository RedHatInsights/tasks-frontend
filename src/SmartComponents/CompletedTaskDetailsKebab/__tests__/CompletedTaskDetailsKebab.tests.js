import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';

import CompletedTaskDetailsKebab from '../CompletedTaskDetailsKebab';

describe('CompletedTaskDetailsKebab', () => {
  let props;

  beforeEach(() => {
    props = {
      status: 'Completed',
      setModalOpened: jest.fn(),
    };
  });

  it('should render correctly', async () => {
    const { asFragment } = render(
      <MemoryRouter keyLength={0}>
        <CompletedTaskDetailsKebab {...props} />
      </MemoryRouter>
    );

    await waitFor(() =>
      userEvent.click(screen.getByLabelText('Task details menu toggle'))
    );
    expect(screen.getByTestId('delete-task-kebab-button')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render Delete disabled', async () => {
    props.status = 'Running';
    const { asFragment } = render(
      <MemoryRouter keyLength={0}>
        <CompletedTaskDetailsKebab {...props} />
      </MemoryRouter>
    );

    await waitFor(() =>
      userEvent.click(screen.getByLabelText('Task details menu toggle'))
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call setModalOpened', async () => {
    render(
      <MemoryRouter keyLength={0}>
        <CompletedTaskDetailsKebab {...props} />
      </MemoryRouter>
    );

    await waitFor(() =>
      userEvent.click(screen.getByLabelText('Task details menu toggle'))
    );
    await waitFor(() =>
      userEvent.click(screen.getByLabelText('delete task menu item'))
    );
    expect(props.setModalOpened).toHaveBeenCalled();
  });
});
