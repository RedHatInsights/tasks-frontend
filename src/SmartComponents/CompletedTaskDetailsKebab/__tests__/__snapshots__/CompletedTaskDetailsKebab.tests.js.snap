// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompletedTaskDetailsKebab should render Delete disabled 1`] = `
<DocumentFragment>
  <button
    aria-expanded="true"
    aria-label="Task details menu toggle"
    class="pf-v5-c-menu-toggle pf-m-expanded pf-m-plain"
    data-ouia-component-id="OUIA-Generated-MenuToggle-plain-2"
    data-ouia-component-type="PF5/MenuToggle"
    data-ouia-safe="true"
    id="executed-task-kebab"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="pf-v5-svg"
      fill="currentColor"
      height="1em"
      role="img"
      viewBox="0 0 192 512"
      width="1em"
    >
      <path
        d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
      />
    </svg>
  </button>
  <div
    class="pf-v5-c-menu"
    data-ouia-component-id="OUIA-Generated-Dropdown-2"
    data-ouia-component-type="PF5/Dropdown"
    data-ouia-safe="true"
    data-popper-escaped="true"
    data-popper-placement="bottom-end"
    data-popper-reference-hidden="true"
    style="position: absolute; left: 0px; top: 0px; z-index: 9999; opacity: 1; transition: opacity 0ms cubic-bezier(.54, 1.5, .38, 1.11); min-width: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="pf-v5-c-menu__content"
    >
      <ul
        class="pf-v5-c-menu__list"
        role="menu"
      >
        <li
          class="pf-v5-c-menu__list-item pf-m-disabled"
          data-ouia-component-id="delete-task-dropdown-item"
          data-ouia-component-type="PF5/DropdownItem"
          data-ouia-safe="true"
          data-testid="delete-task-kebab-button"
          role="none"
        >
          <button
            aria-label="delete task menu item"
            class="pf-v5-c-menu__item"
            disabled=""
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <span
              class="pf-v5-c-menu__item-main"
            >
              <span
                class="pf-v5-c-menu__item-text"
              >
                Delete
              </span>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CompletedTaskDetailsKebab should render correctly 1`] = `
<DocumentFragment>
  <button
    aria-expanded="true"
    aria-label="Task details menu toggle"
    class="pf-v5-c-menu-toggle pf-m-expanded pf-m-plain"
    data-ouia-component-id="OUIA-Generated-MenuToggle-plain-1"
    data-ouia-component-type="PF5/MenuToggle"
    data-ouia-safe="true"
    id="executed-task-kebab"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="pf-v5-svg"
      fill="currentColor"
      height="1em"
      role="img"
      viewBox="0 0 192 512"
      width="1em"
    >
      <path
        d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
      />
    </svg>
  </button>
  <div
    class="pf-v5-c-menu"
    data-ouia-component-id="OUIA-Generated-Dropdown-1"
    data-ouia-component-type="PF5/Dropdown"
    data-ouia-safe="true"
    data-popper-escaped="true"
    data-popper-placement="bottom-end"
    data-popper-reference-hidden="true"
    style="position: absolute; left: 0px; top: 0px; z-index: 9999; opacity: 1; transition: opacity 0ms cubic-bezier(.54, 1.5, .38, 1.11); min-width: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="pf-v5-c-menu__content"
    >
      <ul
        class="pf-v5-c-menu__list"
        role="menu"
      >
        <li
          class="pf-v5-c-menu__list-item"
          data-ouia-component-id="delete-task-dropdown-item"
          data-ouia-component-type="PF5/DropdownItem"
          data-ouia-safe="true"
          data-testid="delete-task-kebab-button"
          role="none"
        >
          <button
            aria-label="delete task menu item"
            class="pf-v5-c-menu__item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            <span
              class="pf-v5-c-menu__item-main"
            >
              <span
                class="pf-v5-c-menu__item-text"
              >
                Delete
              </span>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
